File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/message/digitaltwin/GetPropertyElementListResponse.java
package org.opcfoundation.webserver.types.message.digitaltwin;

import org.opcfoundation.webserver.types.PropertyElementDescriptor;

import java.util.List;

public class GetPropertyElementListResponse {
    private final List<PropertyElementDescriptor> elements;
    private final boolean containsMoreData;

    public GetPropertyElementListResponse(
            List<PropertyElementDescriptor> elements,
            boolean containsMoreData)
    {
        this.elements = elements;
        this.containsMoreData = containsMoreData;
    }

    public List<PropertyElementDescriptor> getElements() {
        return elements;
    }

    public boolean containsMoreData()
    {
        return containsMoreData;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/GetPropertyElementListResponse.ts
import { PropertyElementDescriptor } from "../../../types/digitaltwin/PropertyElementDescriptor";

export class GetPropertyElementListResponse {
    private readonly _elements: Array<PropertyElementDescriptor>;
    private _containsMoreData: boolean;
    private _isProperty: boolean;

    constructor() {
        this._elements = [];
        this._containsMoreData = false;
        this._isProperty = true;
    }

    add(descriptor: PropertyElementDescriptor): void {
        this._elements.push(descriptor);
    }

    get elements(): Array<PropertyElementDescriptor> {
        return this._elements;
    }

    get containsMoreData(): boolean {
        return this._containsMoreData;
    }

    set containsMoreData(containsMoreData: boolean) {
        this._containsMoreData = containsMoreData;
    }

    get isProperty(): boolean {
        return this._isProperty;
    }

    set isProperty(isProperty: boolean) {
        this._isProperty = isProperty;
    }
}